scrapy/core/scheduler.py

```diff
@@ -132,3 +141,27 @@ def _dqdir(self, jobdir):
             if not exists(dqdir):
                 os.makedirs(dqdir)
             return dqdir
+
+
+class Scheduler(BaseScheduler):
+    """
+    Key is the priority of the request (an integer)
+    """
+
+    def request_key(cls, request):
+        return -request.priority
+
+
+class DomainScheduler(BaseScheduler):
+    """
+    Key is the domain and we round robin.  The pqclass parameter is ignored.
+    """
+
+    def __init__(self, dupefilter, jobdir=None, dqclass=None, mqclass=None,
+                 logunser=False, stats=None, pqclass=None):
+        super(DomainScheduler, self).__init__(dupefilter, jobdir=jobdir, dqclass=dqclass,
+                                              mqclass=mqclass, logunser=logunser,
+                                              stats=stats, pqclass=RoundRobinQueue)
+
+    def request_key(cls, request):
+        return urlparse(request.url).netloc
```

Comment:
If we go with round robin, it should be probably based on `request.meta['download_slot']`, or use the same algorithm. 

I also think that's good to have it overridable via request.meta, e.g. in https://github.com/TeamHG-Memex/linkdepth/blob/master/queues.py#L21 I used `request.meta['scheduler_slot']`. This allows to tweak these slots from spiders, and also allows to enable round-robin queue by default: scheduler_slot is empty, so all requests go to the same queue, keeping existing behavior; users who need round-robin can set a scheduler slot per-request.

Reply:
Thanks!  Just implemented.